zenoh/src/lib.rs

```diff
@@ -364,26 +355,58 @@ pub mod config {
     pub use zenoh_config::*;
 }
 
-#[doc(hidden)]
-#[cfg(all(feature = "unstable", feature = "plugins"))]
-pub mod plugins {
-    pub use crate::api::plugins::{
-        PluginsManager, Response, RunningPlugin, RunningPluginTrait, ZenohPlugin, PLUGIN_PREFIX,
-    };
-}
+#[cfg(all(feature = "internal", not(feature = "unstable")))]
+compile_error!(
+    "All internal functionality is unstable. The `unstable` feature must be enabled to use `internal`."
+);
+#[cfg(all(
+    feature = "plugins",
+    not(all(feature = "unstable", feature = "internal"))
+))]
+compile_error!(
+    "The plugins support is internal and unstable. The `unstable` and `internal` features must be enabled to use `plugins`."
+);
 
-#[doc(hidden)]
+#[zenoh_macros::internal]
```

Comment:
`session::init` is not in the `internal` module. Is it wanted to have *internal* items spreads across modules?

Reply:
Good point. Maybe we should do it. But this is to be discussed additionally I think. `session::init` is a way to do `session::open` for plugins and it accepts runtime as a parameter. Maybe it should be put into `internal::runtime` mod, maybe not. Let's leave it as is for now

Reply:
`session::init` is not the only one, there is also `query::ReplySample` for example.
I've no particular objection having *internal* items spreads across modules, but then, maybe we should put back root `buffers` module and mark it with `#[zenoh_macros::internal]`.

Reply:
maybe yes, maybe no. I'm agnostic about that. Moving buffers into "internal" was proposed by @Mallets , I had no objections to that. And I think it's better as it immediately made clear that some examples uses internal API, why they should not

Reply:
Ok.